libnavigation-core/src/test/java/com/mapbox/navigation/core/internal/trip/session/MapboxTripSessionTest.kt

```diff
@@ -501,6 +614,25 @@ class MapboxTripSessionTest {
         coVerify { navigator.setRoute(null) }
     }
 
+    @Test
+    fun checksCancelOngoingUpdateNavigatorStatusDataJobsAreCalledWhenARouteIsSet() {
+        tripSession = spyk(
+            MapboxTripSession(
+                tripService,
+                locationEngine,
+                navigatorPredictionMillis,
+                navigator,
+                logger = logger,
+                accessToken = "pk.1234"
+            ),
+            recordPrivateCalls = true
+        )
+
+        tripSession.route = null
+
+        verify(exactly = 2) { tripSession["cancelOngoingUpdateNavigatorStatusDataJobs"]() }
```

Comment:
Not a big fan of testing `private` functions but that was the "better" way I found to add a test around this weird logic https://github.com/mapbox/mapbox-navigation-android/blob/11b71f3172a92aec46d46e8903b2238a0351e03b/libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt#L77-L81 added to prevent sending outdated callbacks to clients if `updateDataFromNavigatorStatus` is called multiple times consecutively.

Tried different things but couldn't `cancel` `updateNavigatorStatusDataJob` jobs externally from `TripSession` `public` APIs - without making `updateNavigatorStatusDataJobs` `internal`. Can you think of a different approach @kmadsen @LukasPaczos?

Reply:
We'd have to mock the jobs that are created and verify that `Job#cancel` was called.

Reply:
> We'd have to mock the jobs that are created and verify that `Job#cancel` was called.

As above-mentioned, we don't have access to the `updateNavigatorStatusDataJobs` list unless we make it `internal` just for testing ¯\\\_(ツ)\_/¯ It's another option but I liked this other better because it doesn't affect "production" code.